src/GameServer.ts

```diff
@@ -0,0 +1,51 @@
+const socketIo = require('socket.io');
+const cors = require('cors');
+import * as express from 'express';
+import { createServer, Server } from 'http';
+import {GameEvent} from './Constants';
+
+// GameServer Properties
+export class GameServer {
+    public static readonly PORT: number = 8080;
+    private _app: express.Application;
+    private server: Server;
+    private io: SocketIO.Server
+    private port: string | number;
+
+    constructor () {
+        this._app = express();
+        this.port = process.env.PORT || GameServer.PORT;
+        this._app.use(cors())
+        this._app.options('*', cors());
+        this.server = createServer(this._app);
+        this.initSocket();
+        this.listen();
+    }
+
+    private initSocket (): void {
+        this.io = socketIo(this.server);
+    }
+
+    private listen (): void {
```

Comment:
As with the comment above, you don't have to explicitly declare this function to having a void type since typescript will see that you are not returning anything from the function and infer this.

That being said, there isn't any harm in including it if you want to